### PR TITLE
GRO-683 - 1 - Add astro, example page, make astro+jekyll live side by side (temporarily)

### PR DESCRIPTION
**Pull request by bpander**
[GRO-683](https://linear.app/metabase/issue/GRO-683/bulk-of-migration) (PT 1)

### Description
Gets astro+jekyll rendering side by side like how we do with the marketing site. Adds an example page that's rendered in astro. The rest of the docs site is unaffected.

This PR is meant to show that we can have astro+jekyll side by side while also demonstrating how some of the edge cases will be handled.

**EXAMPLE PAGE:**
https://gro-683-1-add-astro.dkkh35gceu2po.amplifyapp.com/docs/astro-test/example-collection/example-doc/

**PRE-EXISTING ISSUE:**
Docs PR envs don't handle the trailing slash redirect when you go to say https://master.dkkh35gceu2po.amplifyapp.com/docs/v0.58 so that same issue exists for this PR env. This PR doesn't touch individual docs pages, so that's not a new issue with this PR (i.e. should be fine in production).

**HOW IT'LL WORK WITH DOCS PAGES:**
This follow-up PR https://github.com/metabase/docs.metabase.github.io/pull/1243 switches the individual docs pages to use astro instead of jekyll. All ~9k pages are generated with astro in that PR env. You can compare it against production. Still some kinks to work out, but it's mostly been pixel perfect.

Follow-up PR env: https://gro-683-2-docs.dkkh35gceu2po.amplifyapp.com/docs/latest/

### CODE REVIEW

File: _includes/schema-markup.html [hunk 1]
```diff
@@ -31,4 +31,4 @@
     ]
   }
 </script>
-{% assign page.schema_markup_added = true %} {% endunless %}
```
Comment (bpander) [inline _includes/schema-markup.html:34]: Pretty sure this is a no-op. You can't actually reassign page properties in jekyll.

File: src/layouts/NewDocsLayout.astro [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  };
+  dirname: string;
+};
+
+const { page, dirname } = Astro.props;
+
+/* Do not show breadcrumb on home page */
+const parts = Astro.url.pathname.split("/");
+const showBreadcrumb =
+  parts[1] == "docs" && parts.length == 4 && parts[3] === "index.html";
```
Comment (gantoreno) [inline src/layouts/NewDocsLayout.astro:18]: I hope I'm reading this right, but it seems like `showBreadcrumb` has the value inverted. The condition itself is fine, but Jekyll does:

Image

Which is basically assigning the opposite value of what the condition evaluates to

Comment (bpander) [reply src/layouts/NewDocsLayout.astro:18]: Yeah I beefed it in this PR. I corrected it in the PR where it actually matters though. https://github.com/metabase/docs.metabase.github.io/pull/1243/changes#diff-43fea136ca01895cbd79ec383fbfa857dae9649fa291efdc0e64223d9df62716R17
